cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -60,8 +60,8 @@ func (u *Upgrade) CanUpgradeEtcd() bool {
 type ClusterState struct {
 	// KubeVersion describes the version of the Kubernetes API Server, Controller Manager, Scheduler and Proxy.
 	KubeVersion string
-	// DNSType describes the type of DNS add-on used in the cluster.
-	DNSType kubeadmapi.DNSAddOnType
+	// DNSKind describes the type of DNS add-on used in the cluster.
+	DNSKind string
```

Comment:
type was fine IMO, added comments above about kind vs type.

Reply:
I am having some long term concerns here. What if at some point we move to cluster addons here and we are passing a GVK with a group fixed for the cluster addons project? We'll end up with different kinds, that will sound something similar to the component name they are deploying.
Hence, that caused the change from type to kind.

Reply:
IMO if the cluster-addons project integrates with kubeadm it's config will not be embeded in the kubeadm-config. this was what i gathered from initial discussions.